ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -980,10 +980,16 @@ class JITModBuilder {
 
         if (isa<CallInst>(I)) {
           if (isa<IntrinsicInst>(I)) {
-            // `llvm.lifetime.start.*` and `llvm.lifetime.end.*` are not really
-            // calls, but rather left-in annotations used in LLVM optimisation
-            // passes. Since these "calls" generate no machine code, nor have
-            // program semantics, we can simply ignore them.
+            // `llvm.lifetime.start.*` and `llvm.lifetime.end.*` are
+            // annotations added by some compiler front-ends to allow backends
+            // to perform stack slot optimisations.
+            //
+            // They are not really calls, and since they generate no machine
+            // code, nor have program semantics, we can simply ignore them.
+            //
+            // OPT: Consider leaving the annotations in, or generating our own
```

Comment:
There may just be a pass we can run over our trace to generate them. Could worth having a look at some point.

Reply:
Yeah, I think generating our own could give more tight bounds. It's an optimisation, so low priority for now.

Reply:
Agreed.